### PR TITLE
ui: add wizard shell structure

### PR DESCRIPTION
**Pull request by ebelo**
## Summary

Add a reusable wizard shell widget for the #609 dock redesign without replacing the current dock or implementing the full wizard flow.

## Changes

- Add `WizardShell` with the target structural chrome: `StepperBar`, separator, scrollable `QStackedWidget`, and compact footer bar.
- Keep page content external so future wizard pages can migrate one at a time without entrenching the current long-scroll dock.
- Add pure Qt-fake tests covering the shell object names, layout order, empty initial page stack, footer updates, and page selection helpers.

## Testing

- `python3 -m pytest tests/ -x -q --tb=short`
- `python3 scripts/package_plugin.py`

Refs #608
Refs #609

### CODE REVIEW

File: ui/dockwidget/wizard_shell.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
+from __future__ import annotations
+
+from typing import Sequence
+
+from .stepper_bar import STEPPER_LABELS, StepperBar, _import_qt_module
```
[the remark this answers is not in the document]
Comment (ebelo) [reply ui/dockwidget/wizard_shell.py]: Addressed in commit 48f86ff: moved the Qt import fallback helper into shared qfit.ui.dockwidget._qt_compat and import the public helper from both stepper_bar and wizard_shell.

File: ui/dockwidget/wizard_shell.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        separator.setFixedHeight(1)
+        return separator
+
+    def _build_content_scroll(self):
+        scroll = QScrollArea(self)
+        scroll.setObjectName("qfitWizardContentScroll")
+        scroll.setWidgetResizable(True)
+        if hasattr(QFrame, "NoFrame") and hasattr(scroll, "setFrameShape"):
+            scroll.setFrameShape(QFrame.NoFrame)
+        return scroll
```
[the remark this answers is not in the document]
Comment (ebelo) [reply ui/dockwidget/wizard_shell.py:106]: Addressed in commit 48f86ff: removed the asymmetric guard and set QFrame.NoFrame directly, matching the separator frame constants.

File: tests/test_wizard_shell.py [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+        second_page = _FakeWidget()
+
+        self.assertEqual(shell.add_page(first_page), 0)
+        self.assertEqual(shell.add_page(second_page), 1)
+        shell.set_step_states(["done", "current", "upcoming", "locked", "locked"])
+        shell.set_current_step(1)
+
+        self.assertEqual(shell.page_count(), 2)
+        self.assertEqual(shell.stepper_bar.states(), ("upcoming", "current", "upcoming", "upcoming", "upcoming"))
+        self.assertEqual(shell.pages_stack.currentIndex(), 1)
```
[the remark this answers is not in the document]
Comment (ebelo) [reply tests/test_wizard_shell.py:317]: Addressed in commit 48f86ff: added an assertion immediately after set_step_states so the test covers delegated state rendering before page selection changes the current step.
